### DISCUSSION TITLE
github-pages 226 | Error:  Liquid syntax error (line 5): Unknown tag 'twitter'

### DISCUSSION THREAD

**Discussion by CryoSky** [alshedivat/al-folio#864] [Q&A] [answered]
**Acknowledge the following**
- [ ] I carefully read and followed the [Getting Started](https://github.com/alshedivat/al-folio#getting-started) guide.
- [ ] I read through [FAQ](https://github.com/alshedivat/al-folio#faq) and searched through the [past issues](https://github.com/alshedivat/al-folio/issues), none of which addressed my issue.
- [ ] The issue I am raising is a potential bug in al-folio and not just a usage question. <br> [For usage questions, please post in the [Discussions](https://github.com/alshedivat/al-folio/discussions) instead of raising an issue.]

**Describe the bug**
Hello, I met the same problem as https://github.com/alshedivat/al-folio/issues/691 this issue as. To avoid the problem, I comment out `mini_racer` and installed Node.Js on my local machine. When I run the command `bundle exec jekyll serve --no-watch`, It gives me a warning that `Jekyll Diagrams: Command Not Found: mmdc` and visit http://127.0.0.1:4000/al-folio/ failed. Then I tried to deploy the page into GitHub without any changes except for the _config.yaml. It returns this error `github-pages 226 | Error:  Liquid syntax error (line 5): Unknown tag 'twitter'` and the action failed so no website is deployed. I'm wondering could anyone give some suggestions?

**To Reproduce**

**Expected behavior**

**Screenshots**

**System (please complete the following information):**
 - OS: Windows Subsystem (WSL 2)
 - Browser (and its version) chrome
 - Jekyll version 4.2.2
- Ruby version 2.7.0p0

**Additional context**

**Comment by ternava** [comment]
I had the same issue/error. [The solution that forked for me is given in this comment.](https://github.com/alshedivat/al-folio/issues/529#issuecomment-1014110681)

**Comment by rohandebsarkar** [comment]
This is probably because you are trying to build the site from `master` and not `gh-pages` branch.

As the @ternava mentioned, change the settings of your repository. In the Pages section, set the branch to `gh-pages`.

**Comment by sayadennis** [comment]
Hi folks, 

I'm experiencing the same error and still a bit confused about the solution. Apologies in advance for my basic questions as I am still new to this. 

In the [README.md](https://github.com/alshedivat/al-folio/blob/master/README.md), I'm following the steps under **For personal and organization webpages** and **To enable automatic deployment**. The step at which I see the `Error: Liquid syntax error` is at the first **Deploy** action, meaning step # 2-3 under **To enable automatic deployment**. The branch `gh-pages` is only created after step # 3 under **To enable automatic deployment**, so I cannot change my branch to `gh-pages` before my first deply. Should I be manually creating the `gh-pages` branch before I `bundle exec`? 

Thanks for your help!

**Comment by rohandebsarkar** [comment]
The `deploy` action should be triggered as soon as you commit to the `master` branch (given you have turned on GitHub Actions). This will automatically create a `gh-pages` branch and commit the built website. Then you can turn on GitHub Pages on the `gh-pages` branch.

**Comment by arashgmn** [comment]
I'm a novice in site deployment, yet I think the 34k lines for building the site (as reported in GitHub actions) are a bit too much. Also, it seems most of it the build is somehow related to `assets/img*` files. Does it ring any bell? (PS: @ternava 's suggestion didn't work for me on Ubuntu 18.04).

**Comment by CryoSky** [comment]
Thank you I think I have fixed this.

**Comment by vdivakar** [comment, marked as answer]
@sayadennis I had the same issue. 
Fix: go to repo Settings -> Pages and select the branch as **None**. Make a small change and push your code again. This would trigger a build and it should succeed this time. You should see the `gh-pages` branch now. So, go back to the Settings->Pages and select the `gh-pages`

**Reply by sayadennis** [reply]
Thank you so much @vdivakar !! This worked for me.

**Reply by jstulips** [reply]
Alternatively, you could start the repo in Private, and it would safely deploy as well. Then switch it to Public, and select `gh-pages` at the Pages branch setting. Just solved this thanks to this discussion. This should really be in the instructions for using the repo and deploying.

**Reply by jeevanchaps** [reply]
Both of the techniques didn't work for me but the solution proposed by @halixness  worked for me and is the easiest one

**Comment by halixness** [comment]
This solution is weak. Solved it via `Settings`, `Pages`, `Build and deployment`, `Source`: `Github Actions (beta)`, which includes jekyll already.

**Reply by jeevanchaps** [reply]
This is the best solution by far.

**Reply by manuelsh** [reply]
this would require to create a jekyll.yml file, does the default work already?
